### DISCUSSION TITLE
Heltec V2 Battery

### DISCUSSION THREAD

**Discussion by t-huyeng** [cyberman54/ESP32-Paxcounter#938] [Q&A]
Hey, 

I have some Heltec LoRa V2 boards which I am using. 

I have also enabled battery mode, but am not sure if I have done something wrong. 
I am using the `halfile = heltecv21.h`, the battery values start at about 2600 [?] and go down. So I think the board is measuring something. But I cannot translate it to a voltage. Is my decoder old or faulty? 

![image](https://user-images.githubusercontent.com/39559223/217902781-403e2676-e424-41a2-8e7f-ed5c37f1cee1.png)

Has anyone had the same problem with Heltec boards?

Thanks

**Comment by cyberman54** [comment]
Heltec has simple battery monitoring by using a voltage divider at GPIO pin. The ratio R1/R2 is setup in heltecv2.h, please check the resistor ohm values. Maybe Heltec changed it on the board, or ratio value in hal file is wrong.

**Reply by t-huyeng** [reply]
Good tip!

I will try to find the resistors on the board and check them. 
A quick look at the schematics shows 220/100 as it is currently in the hal file. https://github.com/Heltec-Aaron-Lee/WiFi_Kit_series/blob/master/SchematicDiagram/WiFi_LoRa_32(V2)/WIFI_LoRa_32_V2(868-915).PDF (5 years old)

This schematics are up to date with PIN 37 https://resource.heltec.cn/download/WiFi_LoRa_32/V2.1/WIFI_LoRa_32_V2.1(868-915).PDF

But it looks more like it needs `#define BAT_VOLTAGE_DIVIDER 3.2 ` to calculate correctly, so the resistors are more like 320/100 I think.

**Reply by cyberman54** [reply]
Note GPIO21 om Heltec v2.1 ("Vext" Switch): It is necessary to switch Vext on, before measuring battery voltage. Otherwise the voltage divider is not powered. This is coded [here](https://github.com/cyberman54/ESP32-Paxcounter/blob/master/src/main.cpp#L170), should work, but maybe to be re-tested.

**Reply by t-huyeng** [reply]
With the tools I have I am currently not able to check the resistors on the board. But I will try to find a way to check them . As described `#define BAT_VOLTAGE_DIVIDER 3.2` works as a quick fix for me at the moment. But I will also try to check if Vext is switched on.
